framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1617,6 +1652,8 @@ protected void applyQualifierParameterDefaults(
                 return;
         }
 
+        applyLocalVariableQualifierParameterDefaults(elt, type);
```

Comment:
Move this call to line 1666 and pass in `tops` rather than recomputing it.   Then in `applyLocalVariableQualifierParameterDefaults` you don't need to check that type has a qualifier parameter.

Reply:
The `tops` variable seems to store qualifier parameter hierarchies for the enclosing class of the local variable, rather than on the type of the local variable itself. So in
```java
ClassWithQualParam1 {
    void m() {
        ClassWithQualParam2 local;
    }
}
```
We would need to apply the defaults to `local` regardless of the qualifier parameters on ClassWithQualParam1. I think this means moving it might be incorrect, because the enclosing class might not have a qualifier parameter at all and we would still apply the new defaults.